test/integration/dialects/postgres/query-interface.test.js

```diff
@@ -15,6 +17,182 @@ if (dialect.match(/^postgres/)) {
       this.queryInterface = this.sequelize.getQueryInterface();
     });
 
+    describe('createFunction', function () {
+
+      it('creates a stored procedure', function () {
+        var self = this;
+        var body = 'return test;';
+        var options = {};
+
+        //setup - make sure we don't have a pre-existing function called create_job
+        return new Promise.method( function () {
+          return self.queryInterface.dropFunction('create_job',[{type:'varchar',name:'test'}])
+          //suppress errors here. if create_job doesn't exist thats ok.
+          .catch( function (err) {  });
+        })()
+        //make our call
+        .then( function () {
+          return self.queryInterface.createFunction('create_job', [{type:'varchar',name:'test'}], 'varchar', 'plpgsql', body, options);
+        })
+        //validate
+        .then( function () { 
+          return self.sequelize.query('select create_job(\'test\');', { type: self.sequelize.QueryTypes.SELECT })
+          .then( function (res) {
+            if (res[0].create_job !== 'test') throw new Error('expected output: test');
+          });
+        })
+        //cleanup
+        .then( function () {
+          return self.queryInterface.dropFunction('create_job',[{type:'varchar',name:'test'}]);
+        });
+      });
+
+      it('treats options as optional', function () {
+        var self = this;
+        var body = 'return test;';
+
+        //setup - ensure we don't have a create_job when we start
+        return new Promise.method( function (){
+          return self.queryInterface.dropFunction('create_job',[{type:'varchar',name:'test'}])
```

Comment:
setup and cleanup both should go in `beforeEach` and `afterEach`. But you don't need to throw any errors inside `beforeEach` or `afterEach`